var/spack/repos/builtin/packages/py-salib/package.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PySalib(PythonPackage):
+    """Python implementations of commonly used sensitivity analysis methods."""
+
+    homepage = "http://salib.readthedocs.org"
+    pypi     = "SALib/SALib-1.4.0.1.tar.gz"
+
+    maintainers = ['schmitts']
+
+    version('1.4.0.1', sha256='dbf6e865af9f3be82a79cf64889ed66d6d3b6803f0c22a242a112876789d49e7')
+
```

Comment:
Missing build dep on setuptools 38.3+

Reply:
Missing build dep on pyscaffold 3.2.0:3.2.999

Reply:
Wheel might also be being pulled in, at least at build time. You'll have to check the build log.

Reply:
Tests don't pass when I don't have the wheel explicitly in.

Reply:
scaffold was only used to create the package and is not used after

Reply:
Which tests don't pass? Are the dependencies in `setup.cfg` not automatically being downloaded by setuptools? Also, setuptools should be build/run, not build.

Reply:
> Which tests don't pass? Are the dependencies in `setup.cfg` not automatically being downloaded by setuptools? Also, setuptools should be build/run, not build.

When I run tests from the source with pytest and don't load explicitely the wheel package:

```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../spack/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/python-3.8.10-fnsjttcwiaivv4ftqiiifrrmpftk2xfh/lib/python3.8/subprocess.py:415: in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

input = None, capture_output = False, timeout = None, check = True, popenargs = (['salib', 'sample', 'saltelli', '-p', './src/SALib/test_functions/params/Ishigami.txt', '-o', ...],), kwargs = {'stderr': -2, 'stdout': -1}
process = <subprocess.Popen object at 0x7fd27add6850>
stdout = b'Traceback (most recent call last):\n  File "/home/sschmitt/spack/opt/spack/linux-ubuntu20.04-haswell/gcc-9.3.0/py-sa...q, requirers)\npkg_resources.DistributionNotFound: The \'wheel\' distribution was not found and is required by SALib\n', stderr = None
retcode = 1
```

Reply:
Should I add a variant "tests" that pulls in `wheel` and `py-pytest` and `py-pytest-cov`.

Reply:
Nah, we generally don't add variants to packages unless they change what gets installed to the installation prefix. We added a `type='test'` dependency type for this purpose. However, in this case, these tests aren't run by the Spack package, so there's no reason to add them. Unless setuptools tries to download and install wheel itself because it isn't found and is listed in `install_requires`.